packages/vite/src/node/server/transformRequest.ts

```diff
@@ -142,6 +142,12 @@ export async function transformRequest(
     isDebug && debugTransform(`${timeFrom(transformStart)} ${prettyUrl}`)
     code = transformResult.code!
     map = transformResult.map
+
+    transformResult.watchFiles.forEach((file) => {
+      if (mod.id) {
+        moduleGraph.addWatchModuleToFile(file, mod.id)
+      }
+    })
```

Comment:
**suggestion:** Looks like we can move the condition out of the loop :thinking:  
Otherwise that really looks like `addWatchModuleToFile` has an ugly side effect

Reply:
Good catch. Fail on my part.